src/i2c/src/vhu_i2c.rs

```diff
@@ -183,16 +206,23 @@ impl<D: I2cDevice> VhostUserI2cBackend<D> {
 
         for (i, desc_chain) in requests.iter().enumerate() {
             let descriptors: Vec<_> = desc_chain.clone().collect();
-            let desc_buf = descriptors[1];
-            let desc_in_hdr = descriptors[2];
+            let desc_in_hdr = descriptors[descriptors.len() - 1];
             let mut len = size_of::<VirtioI2cInHdr>() as u32;
 
-            // Write the data read from the I2C device
-            if desc_buf.is_write_only() {
-                desc_chain
-                    .memory()
-                    .write(&reqs[i].buf, desc_buf.addr())
-                    .map_err(|_| Error::DescriptorWriteFailed)?;
+            if descriptors.len() == 3 {
+                let desc_buf = descriptors[1];
+
```

Comment:
so what happens if we don't have 3 descriptors - is that an invalid and unexpected condition?

Reply:
No, it is a valid zero-length operation then, just that we don't need to process the buffer in that case.